src/Match2Cases.jl

```diff
@@ -375,28 +366,28 @@ function build_state_machine_core(
         code = pop!(work_queue)
         if code.action isa Nothing
             set_next!(code, state)
-            @assert !(code.action isa Nothing)
+            @assert code.action !== nothing
```

Comment:
It's generally preferred w/ `nothing` to compare the _value_ instead of the type and to compare using `===`/`!==` because it's a singleton. I don't think it much matters any more to the compiler in terms of efficiency, but it's become the norm in any case.